aten/src/ATen/core/boxing/impl/boxing.h

```diff
@@ -5,75 +5,247 @@
   }
+};
 
-  TORCH_INTERNAL_ASSERT(false, "Tried to call KernelFunction::call() for a kernel that only has a boxed kernel and doesn't support calling from an unboxed API yet.");
-}
+// at::Quantizer
+template <class... Args>
+using has_quantizer_arg =
+  guts::disjunction<
+    std::is_same<at::Quantizer, std::decay_t<Args>>...,
+    std::is_same<c10::intrusive_ptr<at::Quantizer>, std::decay_t<Args>>...
+  >;
 
-// SFINAE version for ops with returns
 template<class Result, class... Args>
-std::enable_if_t<supports_boxing<Result, Args...>::value && !std::is_same<void, Result>::value, Result>
-boxAndCallBoxedFunc(KernelFunction::InternalBoxedKernelFunction* boxed_kernel_func, OperatorKernel* functor, const OperatorHandle& opHandle, Args... args) {
-  // TODO Reuse stack vector instead of allocating?
-  torch::jit::Stack stack;
-  torch::jit::push(stack, std::forward<Args>(args)...);
-
-  (*boxed_kernel_func)(functor, opHandle, &stack);
-
-  TORCH_INTERNAL_ASSERT(stack.size() == 1, "A boxed kernel should only push one return to the stack");
-  return std::move(stack[0]).to<Result>();
-}
+struct BoxedKernelWrapper<Result(Args...), std::enable_if_t<has_quantizer_arg<Args...>::value, void>> {
+  static Result call(KernelFunction::InternalBoxedKernelFunction*, OperatorKernel*, const OperatorHandle&, Args... args) {
+    TORCH_INTERNAL_ASSERT(false, "Unboxed call to a boxed kernel with unboxable parameter type at::Quantizer.");
+  }
+};
 
-// SFINAE version for ops without returns
+// 2. Supported signatures, other than ref-passing.
+//
 template<class Result, class... Args>
-std::enable_if_t<supports_boxing<Result, Args...>::value && std::is_same<void, Result>::value, Result>
-boxAndCallBoxedFunc(KernelFunction::InternalBoxedKernelFunction* boxed_kernel_func, OperatorKernel* functor, const OperatorHandle& opHandle, Args... args) {
-  // TODO Reuse stack vector instead of allocating?
-  torch::jit::Stack stack;
-  torch::jit::push(stack, std::forward<Args>(args)...);
-
-  (*boxed_kernel_func)(functor, opHandle, &stack);
+struct BoxedKernelWrapper<
+  Result(Args...),
+  std::enable_if_t<
+    can_box_all<Args...>::value && can_unbox<Result>::value && !is_tuple_of_tensor_refs<Result>::value,
+    void
+  >
+> {
+  static Result call(
+    KernelFunction::InternalBoxedKernelFunction* boxed_kernel_func,
+    OperatorKernel* functor,
+    const OperatorHandle& opHandle,
+    Args... args
+  ) {
+    // TODO Reuse stack vector instead of allocating?
+    torch::jit::Stack stack;
+    stack.reserve(sizeof...(Args));
+    torch::jit::push(stack, std::forward<Args>(args)...);
+
+    (*boxed_kernel_func)(functor, opHandle, &stack);
+
+    return guts::if_constexpr<!std::is_same<void, Result>::value>(
+      [&] (auto delay_check) {
+        TORCH_INTERNAL_ASSERT(
+          stack.size() == 1,
+          "Boxed kernel was expected to push exactly one return value to the stack."
+        );
+        return delay_check(std::move(stack[0]).to<Result>());
```

Comment:
TIL. (I didn't know how to use the delay check idiom for return types. Now I know.....)

Reply:
Ahahaha yeah well basically YIL... but calling it `delay_check` [instead of `_`](https://github.com/pytorch/pytorch/blob/master/c10/util/C++17.h#L242-L251) definitely helps me read the idiom 😬